src/mailer/models.py

```diff
@@ -251,9 +251,12 @@ def log(self, message, result_code, log_message=""):
             log_message=log_message,
         )
 
-    def purge_old_entries(self, days):
+    def purge_old_entries(self, days, result_codes=None):
+        if not result_codes:
```

Comment:
I think this would be better as `if result_codes is None`, because:
* this is what you mean
* there are other things that could be boolean false, for example an empty list. If I were to explicitly pass ` purge_old_entries(5, result_codes=[])`, I would expect that it would purge no entries. It may or may not make sense to call the method this way, and we might not be calling it right now, but we might in the future, and someone else might use it - and we should do the right thing.